### ISSUE TITLE
Add ID attribute to facet modal

### ISSUE THREAD

**Issue by ggeisler** [projectblacklight/arclight#551]
The modal used to show facet values that exceed the limit of values shown in the facet inline currently has an accessibility error, because the modal uses an `aria-labelledby` attribute that references values that never appear as an ID for any element.

To fix this, I believe we should add an ID attribute to the element that contains text we want to use as the label, which I think is the `<h3 class="modal-title">...</h3>` element. To make this as clear as possible, I suggest:

1. Add an `id="facet-modal"` attribute to the `<h3 />` noted above.
2. Add `facet-modal` as a value to the `aria-labelledby` attribute of the element highlighted in the screenshot below:

![Screen Shot 2019-07-24 at 10 30 11 AM](https://user-images.githubusercontent.com/101482/61815389-00036900-adff-11e9-8f2e-c1088e37c5fd.png)

Note that I don't know the history behind the current values of `modal menu` that are currently in the `aria-labelledby` attribute, so am not sure if they are actually needed (since there doesn't appear to be any ID value of either `modal` or `menu` I don't think so, but I'm definitely not certain).

Also, I had expected we would inherit this modal from Blacklight, and we probably do, but as best I can tell from the current demo.projectblacklight instance, the `aria-labelledby` attribute is not present in the version of the facet modal there, so I'm not sure why it appears only in ArcLight (i.e., whether we added it for some specific reason like trying to improve ArcLight accessibility, or there is something else going on with Bootstrap versions that I don't understand). But ultimately it seems like this might be something more appropriate in Blacklight, and ArcLight would just inherit it.

**Comment by jrgriffiniii** [comment]
> Note that I don't know the history behind the current values of modal menu that are currently in the aria-labelledby attribute, so am not sure if they are actually needed (since there doesn't appear to be any ID value of either modal or menu I don't think so, but I'm definitely not certain).

https://github.com/projectblacklight/blacklight/commit/c569c2cf8743bfc2ac1453685b8fae7954034415 and https://github.com/projectblacklight/blacklight/issues/1776 leads me to believe that these values were indeed unnecessary.

**Comment by camillevilla** [comment]
I agree with @jrgriffiniii's analysis here. Thanks for the git archaeology!  
For posterity, the current state of the modal:
```html
<div id="blacklight-modal" class="modal fade show" tabindex="-1" role="dialog" style="padding-right: 15px; display: block;" aria-modal="true">
```
